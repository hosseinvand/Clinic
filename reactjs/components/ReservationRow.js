import React, {Component, PropTypes} from 'react'
import { Router, Route, Link, browserHistory, IndexRoute } from 'react-router'

const  statusToString = {
    PENDING : 'در دست بررسی',
    ACCEPTED : 'تعیین شده',
    REJECTED : 'رد شده',
    EXPIRED : 'تاریخ گذشته'
}

const  statusToClass = {
    PENDING : "bg-info text-info",
    ACCEPTED : "bg-success text-success",
    REJECTED : "bg-danger text-danger",
    EXPIRED : "bg-danger text-danger"
}

class ReservationRow extends Component {
  constructor(props){
    super(props)
  }

  render(){
    const {
        pk,
        doctor_pk,
        speciality,
        date,
        from,
        to,
        status,
        full_name
    } = this.props

    return(
      <tr id={pk} className="centered-cells">
        <td><a href={`/notebook/doctor/${doctor_pk}`} className="text-primary">{ full_name }</a></td>
        <td>{ speciality }</td>
        <td>{ date }</td>
        <td>{ from } تا { to }</td>
        <td className={statusToClass[status]}> { statusToString[status] } </td>
      </tr>
    )
  }
}

ReservationRow.propTypes = {
    pk: PropTypes.number,
    colorClass: PropTypes.string,
    full_name: PropTypes.string,
    doctor_pk: PropTypes.number,
    speciality: PropTypes.string,
    date: PropTypes.string,
    from: PropTypes.number,
    to: PropTypes.number,
    status: PropTypes.string
}

export default ReservationRow
