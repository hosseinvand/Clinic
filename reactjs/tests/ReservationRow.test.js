import React from 'react'
import ReservationRow from '../components/ReservationRow'
import renderer from 'react-test-renderer'

test('renders correctly', () => {
  const tree = renderer.create(
    <ReservationRow date="۱۳۹۴/۲/۲" doctor_pk={2} from={14} full_name="علی رادمنش" pk={3} speciality="چشم" status="PENDING" to={16} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});