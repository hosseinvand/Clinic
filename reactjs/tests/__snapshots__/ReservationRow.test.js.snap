exports[`test renders correctly 1`] = `
<tr
  className="centered-cells"
  id={3}>
  <td>
    <a
      className="text-primary"
      href="/notebook/doctor/2">
      علی رادمنش
    </a>
  </td>
  <td>
    چشم
  </td>
  <td>
    ۱۳۹۴/۲/۲
  </td>
  <td>
    14
     تا 
    16
  </td>
  <td
    className="bg-info text-info">
     
    در دست بررسی
     
  </td>
</tr>
`;
